packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -0,0 +1,174 @@
+const contentful = require(`contentful`)
+const crypto = require(`crypto`)
+const _ = require(`lodash`)
+
+const typePrefix = `contentful__`
+const makeTypeName = type => `${typePrefix}${type.replace(/-/g, `_`)}`
+
+exports.sourceNodes = async (
+  { boundActionCreators, getNode, hasNodeChanged, store },
+  { spaceId, accessToken }
+) => {
+  const {
+    createNode,
+    updateSourcePluginStatus,
+    touchNode,
+  } = boundActionCreators
+  updateSourcePluginStatus({
+    plugin: `gatsby-source-contentful`,
+    status: {
+      ...store.getState().status[`gatsby-source-contentful`],
+      ready: false,
+    },
+  })
+
+  // Touch existing Contentful nodes so Gatsby doesn't garbage collect them.
+  _.values(store.getState().nodes)
+    .filter(n => n.internal.type.slice(0, 8) === typePrefix)
+    .forEach(n => touchNode(n.id))
```

Comment:
Ah, makes sense. I'll update this and test by removing some entries in Contentful.